lib/Sema/CSDiag.cpp

```diff
@@ -4385,6 +4385,21 @@ static bool diagnoseRawRepresentableMismatch(CalleeCandidateInfo &CCI,
   return false;
 }
 
+// Extract expression for failed argument number
+static Expr *getFailedArgumentExpr(CalleeCandidateInfo CCI, Expr *argExpr) {
+  if (auto *TE = dyn_cast<TupleExpr>(argExpr))
+    return TE->getElement(CCI.failedArgument.argumentNumber);
+  else if (auto *PE = dyn_cast<ParenExpr>(argExpr)) {
+    assert(CCI.failedArgument.argumentNumber == 0 &&
+           "Unexpected argument #");
+    return PE->getSubExpr();
+  } else {
+    assert(CCI.failedArgument.argumentNumber == 0 &&
```

Comment:
Just a note (since you've mode this code) - that empty parameters/arguments are represented as an empty tuple, so there are only two choices really - paren or tuple.